app/models/user.rb

```diff
@@ -92,14 +92,37 @@ def neighbourhood_admin_for_partner?(partner_id)
   end
 
   def only_neighbourhood_admin_for_partner?(partner_id)
-    neighbourhood_admin? &&
+    (neighbourhood_admin? || partnership_admin?) &&
       Set.new(owned_neighbourhood_ids).superset?(
         Set.new(
           Partner.find_by(id: partner_id)&.owned_neighbourhood_ids
         )
       )
   end
 
+  def only_partnership_admin_for_partner?(partner_id)
+    return unless partnership_admin?
+
+    partner = Partner.find(partner_id)
+
+    user_neighbourhoods = owned_neighbourhood_ids
+    user_tags = tags.pluck(:id)
+
+    a = Set.new(user_neighbourhoods)
```

Comment:
would be clearer if these variables had semantic names, same for below, (also prefer not repeating variable names within the same scope) but that's just my preference for clarity, so fine if you want to leave as is